What Are PWAs - intro doc (#2049)
GoogleChrome/web.dev: src/site/content/en/progressive-web-apps/what-are-pwas/index.md

> ---

Reviewer: Changing the document title from `what-are-pwas` to `what-are-progressive-web-apps` might give you better SEO (depending on what phrase you want to target), I'm pretty sure that document titles do have a noticeable effect on documentation SEO in general.

title: What are Progressive Web Apps?
authors:
  - samrichard
  - petelepage
date: 2020-01-06
updated: 2020-01-06
description: |
  An introduction to Progressive Web Apps (PWAs) and the three pillars that
  separate them from other web apps.
tags:
  - post
  - progressive-web-apps
draft: true
---

The web is an incredible platform. Its mix of ubiquity across devices and
operating systems, its user-centered security model, and the fact that neither
its specification nor its implementation is controlled by a single company makes
the web a unique platform to develop software on. Combined with its inherent

> linkability, it's possible to search across it and share what you've found with

Reviewer: ```suggestion
linkability, it's possible to search it and share what you've found with
```

anyone, anywhere. Whenever you go to a website, it's up-to-date, and your
experience with that site can be as ephemeral or as permanent as you'd like. Web
applications can reach _anyone, anywhere, on any device_ with a single codebase.

> Native applications, are known for being incredibly rich and reliable. They're

Reviewer: ```suggestion
Native applications are known for being incredibly rich and reliable. They're
```

> ever-present, on home screens and docks and taskbars. They work regardless of

Reviewer: ```suggestion
ever-present, on home screens, docks, and taskbars. They work regardless of
```

network connection. They launch in their own standalone experience. They can
read and write files from the local file system, access hardware connected via

> USB, serial or bluetooth, and even interact with data stored on your device,

Reviewer: ```suggestion
USB, serial, or Bluetooth, and even interact with data stored on your device,
```

like contacts and calendar events. In native applications, you can do things
like take pictures, see playing songs listed on the home screen, or control song
playback while in another app. Native applications feel like _part_ of the
device they run on.

<figure class="w-figure w-figure--center">
  <img src="./capabilities-reach.svg" style="max-width: 100%;"
    alt="A graph illustrating the relative capabilites and reach of native apps, with high capabilities, web apps, with high reach, and progressive web apps, which have both high capabilities and high reach."/>
  <figcaption class="w-figcaption w-figcaption--fullbleed">

>     Capabilities vs reach of native apps, web app, and progressive web apps.

Reviewer: ```suggestion
    Capabilities vs. reach of native apps, web app, and progressive web apps.
```

  </figcaption>
</figure>

If you think about native apps and web apps in terms of capabilities and reach,
native apps represent the best of capabilities whereas web apps represent the
best of reach. So where do Progressive Web Apps fit in?

Progressive Web Apps (PWA) are built and enhanced with modern APIs to deliver
native-like capabilities, reliability, and installability while reaching
_anyone, anywhere, on any device_ with a single codebase.

> ## The three app pillars

Reviewer: I think this one is debatable so I'm just going to leave it be

Progressive Web Apps are web applications that have been designed so they are
capable, reliable, and installable. These three pillars transform them into an
experience that feels like a native application.

### Capable

> The web is quite capable in its own right today. You can build a hyper-local

Reviewer: ```suggestion
The web is quite capable in its own right today. For example, you can build a hyper-local
```

video chat app using WebRTC, geolocation, and push notifications. You can make

> that app installable and take those conversations virtual with WebGL and WebVR.

Reviewer 2: Change 'WebVR' to 'WebXR'. 'WebVR' is specifically the name of an API that's no longer supported. 'WebXR' is the name of the supported API. Alternatively you could use a generic term instead of an API name. In that case, I'd use 'augmented reality' rather than 'virtual reality'. The immersive web folks have fleshed out AR first and consider its use cases the more likely to be useful on the web.

Reviewer: ```suggestion
that app installable and take those conversations virtual with WebGL and WebXR.
```

With the introduction of Web Assembly, developers can tap into other ecosystems,
like C, C++, and Rust, and bring decades of work and capabilities to the web

> too! [Squoosh.app](https://squoosh.app/), for instance, leveraged this for its

Reviewer 2: Change exclamation mark to a period.
Change 'leveraged' to 'leverages'.

Reviewer: ```suggestion
too. [Squoosh.app](https://squoosh.app/), for instance, leverages this for its
```

advanced image compression.

Until recently, only native apps could really lay claim to these capabilities.
While some capabilities are still out of the web's reach, new and upcoming APIs
are looking to change that, expanding what the web can do with features like
file system access, media controls, app badging, and full clipboard support. All
of these capabilities are built with the web's secure, user-centric permission
model, ensuring that going to a website is never a scary proposition for users.

Between modern APIs, Web Assembly, and new and upcoming APIs, web applications
are more capable than ever, and those capabilities are only growing.

### Reliable

A reliable Progressive Web App feels fast and dependable regardless of the
network.

Speed is critical for getting users to _use_ your experience. In fact, as page
load times go from 1 second to ten seconds, the probability of a user bouncing
[increases by 123%](https://www.thinkwithgoogle.com/marketing-resources/data-measurement/mobile-page-speed-new-industry-benchmarks/).

> Performance doesn't stop `onload`. Users should never wonder whether their

Reviewer 2: Performance doesn't stop after the `onload` event.

I know I usually say to remove words. Sometimes it's needed for clarity.

Reviewer: ```suggestion
Performance doesn't stop after the `onload` event. Users should never wonder whether their
```

interaction—for example, clicking a button—was registered or not. Scrolling and
animation should feel smooth. Performance affects your entire experience, from
how users perceive your application to how it actually performs.

Finally, reliable applications need to be usable regardless of network
connection. Users expect apps to start up on slow or flaky network connections
or even when offline. They expect the most recent content they've interacted
with, like media tracks or tickets and itineraries, to be available and usable
even if getting a request to your server is hard. When a request isn't possible,
they expect to be told there's trouble instead of silently failing or crashing.

Users love apps that respond to interaction in the blink of an eye, and an
experience they can depend on.

### Installable

Installed Progressive Web Apps run in a standalone window instead of a browser

> tab. They're launchable from on the user's home screen, dock, taskbar, or shelf.

Reviewer: ```suggestion
tab. They're launchable from the user's home screen, dock, taskbar, or shelf.
```

It's possible to search for them on a device and jump between them with the app
switcher, making them feel like part of the device they're installed on.

> New capabilities get opened up once installed. Keyboard shortcuts usually

Reviewer 2: 'New capabilities get opened up once installed' --> 'New capabilities open up after a web app is installed.'

The flip side of 'omit needless words', is 'do not take shortcuts at the cost of clarity'. The original wording feels a bit like a shortcut.

Reviewer: +1 this sentence sounded funky to me

Reviewer: ```suggestion
New capabilities open up after a web app is installed. Keyboard shortcuts usually
```

> reserved when running in the browser, become available for use. Progressive Web

Reviewer 2: Delete 'for use'.

Reviewer: ```suggestion
reserved when running in the browser, become available. Progressive Web
```

Apps can register to accept content from other applications, or to be the
default application to handle different types of files.

When a Progressive Web App moves out of a tab and into a standalone app window,
it transforms how users think about it and interact with it.

## The best of both worlds

At their heart, Progressive Web Apps are just web applications. Using
progressive enhancement, new capabilities are enabled in modern browsers. Using
service workers and a web app manifest, your web application becomes reliable

> and installable. If they're not available, users still get the core experience.

Reviewer 2: What does the pronoun 'they' refer to?

Reviewer: ```suggestion
and installable. If the new capabilities aren't available, users still get the core experience.
```
